### DISCUSSION TITLE
Add Guard.IsDefined<TEnum>(...) (and maybe more enum related features)

### DISCUSSION THREAD

**Discussion by benjamin-hejl** [CommunityToolkit/dotnet#644] [Ideas]
Hello, I am starting to use CommunityToolkit.Diagnostics, mainly the Guard and ThrowHelper classes but I have noticed lack of enum related checks. Among other things there could be methods like:
```
Guard.IsDefined<TEnum>(TEnum value, ...)
Guard.HasFlag<TEnum>(TEnum value, TEnum flag, ...)
Guard.IsAnyOf<TEnum>(TEnum value, params TEnum[] ...)
Guard.IsNoneOf<TEnum>(TEnum value, params TEnum[] ...)
```
I should be able to implement this feature.

**Comment by michael-hawker** [comment]
FYI @Sergio0694 transferred from WCT discussions. Seems like a nice idea.

@benjamin-hejl what would `IsDefined` do? Wouldn't that be by construction as enums are strongly-typed?

Do you have any usage patterns from examples of APIs you're trying to guard currently?

**Reply by benjamin-hejl** [reply]
It would forward call to `Enum.IsDefined<TEnum>(value)`.

As for usage, I have a struct wrapper around `Int64` with getters which interprets parts of the value as enums using shifts and bit masking. Methods using this struct should check that the values are correct.
